File: JavaScript/javascript-exercises/DONE-calculator/calculator.js
function add (a, b) {
	return a + b
}

function subtract (a, b) {
	return a - b
}

function sum (array) {
	return array.reduce((a,b) => a + b, 0)
}

function multiply (array) {
	return array.reduce((a,b) => a * b)
}

function power(a,b) {
	return a ** b
}

function factorial(num) {
	let result = num
	if(num === 0 || num === 1) {
		return 1
	}
	while(num > 1) {
		num --
		result *= num
	}
	return result
}

module.exports = {
	add,
	subtract,
	sum,
	multiply,
    power,
	factorial
}

File: Ruby/learn_ruby/02_calculator/calculator.rb
#write your code here
def add num1,num2
    num1 + num2
end

def subtract num1,num2
    num1 - num2
end

def sum array
    output = 0

    if array.empty?
        "ERROR! EMPTY ARRAY"
    else
        array.each do |num|
            output = output + num
        end
    end
    output
end

def multiply num1,num2
    num1 * num2
end

def power num1,num2
    num1 ** num2
end

def factorial num
    result = 1

    if num == 0
        1
    else
        while num > 0 do
            result = result * num
            num -= 1
        end
    end
    result
end